### DISCUSSION TITLE
Fresh Install From Master, Why Do I Get Stuck On Login Page While URL in ../Z_UpgradeDatabase.php?

### DISCUSSION THREAD

**Discussion by bcinteractive** [timschofield/webERP#310] [Installation Issues] [answered]
Fresh install from master, why do I get stuck on login page while URL in ...../Z_UpgradeDatabase.php?

**Comment by andrewcouling** [comment]
I also have this issue today with a fresh install.

**Comment by dalers** [comment]
This may be related to my last reply to issue #297 (https://github.com/timschofield/webERP/discussions/297#discussioncomment-11562692)

**Reply by andrewcouling** [reply]
My symptom is slightly different to the one described in #297... I cannot login at all. Have tested with Chrome, Edge and FireFox

**Reply by craigbass76** [reply]
Did you ever figure it out?

**Reply by craigbass76** [reply]
Tim just gave me a new header.php, and while the apache error log is not longer giving me that message, I'm still getting redirected to Z_UpgradeDatabase.php.

**Comment by andrewcouling** [comment]
Here is the web server error log I'm getting which appears to relate to login_data:

`erp-repo.handloomholdings.com [Sat Dec 14 09:04:59 2024] [error] [client 212.56.99.66:0] AH01071: Got error 'PHP message: PHP Fatal error:  Uncaught mysqli_sql_exception: Table 'weberp-repo-353037355684.login_data' doesn't exist in /home/sites/3a/2/2b806fa8d9/public_html/erp-repo/includes/ConnectDB_mariadb.inc:58\nStack trace:\n#0 /home/sites/3a/2/2b806fa8d9/public_html/erp-repo/includes/ConnectDB_mariadb.inc(58): mysqli_query(Object(mysqli), 'SELECT sessioni...')\n#1 /home/sites/3a/2/2b806fa8d9/public_html/erp-repo/includes/session.php(187): DB_query('SELECT sessioni...')\n#2 /home/sites/3a/2/2b806fa8d9/public_html/erp-repo/index.php(4): include('/home/sites/3a/...')\n#3 {main}\n  thrown in /home/sites/3a/2/2b806fa8d9/public_html/erp-repo/includes/ConnectDB_mariadb.inc on line 58'`

**Comment by timschofield** [comment]
The trouble is I'm not getting this problem. I have however done some
changes to:

install/index.php
includes/session.php
includes/header.php

I am attaching them here for people to test before I commit them.

Tim


On Wed, 18 Dec 2024 at 19:42, Craig Parker ***@***.***> wrote:

> Tim just gave me a new header.php, and while the apache error log is not
> longer giving me that message, I'm still getting redirected to
> Z_UpgradeDatabase.php.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/timschofield/webERP/discussions/310#discussioncomment-11609738>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAL6LE4X37XFAVZXBXUVL32GHF37AVCNFSM6AAAAABTTA2ERGVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCNRQHE3TGOA>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***
> com>
>


-- 
www.weberp.org <http://www.weberpafrica.com>
@TimSchofield2
Blog: https://kwamoja.home.blog/

**Comment by craigbass76** [comment]
Does it matter that I'm using MariaDB? I've unzipped the new version, stuck these files where they go, and still not up and running.

I did notice that whatever I stuck in for a db name, my apache error logs kept saying weberpdemo wasn't available. When I  set up another instance and created a db called weberpdemo, that error disappeared.

I did try commenting out the if loop (around line 236 in session.php) and didn't get any farther.

Except wait! I just tried commenting that out again and now I'm sitting at a login screen. I'm going to go wipe/reinstall before I get too excited...

**Comment by timschofield** [comment]
Craig, when you sent me a link earlier I was able to log in, so your
install is working, but it's that link at the end of the install to restart
that is failing.

On Wed, 18 Dec 2024 at 21:12, Craig Parker ***@***.***> wrote:

> Does it matter that I'm using MariaDB? I've unzipped the new version,
> stuck these files where they go, and still not up and running.
>
> I did notice that whatever I stuck in for a db name, my apache error logs
> kept saying weberpdemo wasn't available. When I set up another instance and
> created a db called weberpdemo, that error disappeared.
>
> I did try commenting out the if loop (around line 236 in session.php) and
> didn't get any farther.
>
> Except wait! I just tried commenting that out again and now I'm sitting at
> a login screen. I'm going to go wipe/reinstall before I get too excited...
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/timschofield/webERP/discussions/310#discussioncomment-11610413>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAL6LC33HJ2YOJYLX5ZQN32GHQK3AVCNFSM6AAAAABTTA2ERGVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCNRRGA2DCMY>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
www.weberp.org <http://www.weberpafrica.com>
@TimSchofield2
Blog: https://kwamoja.home.blog/

**Reply by craigbass76** [reply]
Weird. I was doing  a hard refresh, and got no love. With those lines commented out, I get the login page though. I'm setting it up in another sub of /var/www/html on a new db to see what happens. I just had an elated swearing fit, so I'm ready to sit back down again.

**Reply by craigbass76** [reply]
Ok, here are my exact steps - this is on a headless Ubuntu 22.04 box, and the three`cp` commands are those files you emailed earlier today:

```
unzip webERP-master.zip
sudo chown -R www-data:www-data webERP-master
cp ~/index.php webERP-master/install/
cp ~/session.php webERP-master/includes/
cp ~/header.php webERP-master/includes/

MariaDB > create database erp;
```

Start install in a browser
 - All tests passed
 Check Install demo data, let install complete
Restart weberp

Taken to http://serverIP/webERP-master/Logout.php?Installed=Yes

Page isn't working message
```
 Apache error log sayeth: [Wed Dec 18 22:15:17.523041 2024] [php:error] [pid 26476] [client clientIP:57474] PHP Fatal error:  Uncaught mysqli_sql_exception: Unknown database 'weberpdemo' in /var/www/html/webERP-master/includes/ConnectDB_mariadb.inc:11\nStack trace:\n#0 /var/www/html/webERP-master/includes/ConnectDB_mariadb.inc(11): mysqli_connect()\n#1 /var/www/html/webERP-master/includes/ConnectDB.inc(29): include_once('...')\n#2 /var/www/html/webERP-master/includes/session.php(54): include('...')\n#3 /var/www/html/webERP-master/Logout.php(3): include('...')\n#4 {main}\n  thrown in /var/www/html/webERP-master/includes/ConnectDB_mariadb.inc on line 11, referer: http://serverIP/webERP-master/install/index.php?Page=6
```

DB name I set up was simply erp. I have to use weberpdemo, or the script fails.

Tried visiting http://serverIP/webERP-master/ - nada

Starting over...

Same commands as before, but now...
```
create database weberpdemo;
```

Go through web browser install

Restart webERP -- not working

Apache error log says:
```
 PHP Fatal error:  Uncaught mysqli_sql_exception: Table 'weberpdemo.login_data' doesn't exist in /var/www/html/webERP-master/includes/ConnectDB_mariadb.inc:58\nStack trace:\n#0 /var/www/html/webERP-master/includes/ConnectDB_mariadb.inc(58): mysqli_query()\n#1 /var/www/html/webERP-master/Logout.php(9): DB_query()\n#2 {main}\n  thrown in /var/www/html/webERP-master/includes/ConnectDB_mariadb.inc on line 58, referer: http://serverIP/webERP-master/install/index.php?Page=6
```

It's been wiped a couple times, but still sitting up at http://18.118.31.32/webERP-master/

**Comment by andrewcouling** [comment]
> I am attaching them here for people to test before I commit them.

@timschofield, I can't see the attachments which you mention above. Can you try attaching them again or send them by email please?

**Reply by craigbass76** [reply]
@timschofield are they newer versions of what you gave me the other day?

**Comment by bcinteractive** [comment]
Just an update: from Master (as of now)  here is the error I got - doing an fresh install
`
PHP Fatal error:  Uncaught mysqli_sql_exception: You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near 'INSERT INTO `tags` VALUES (0, 'None')' at line 7 in /home/.../webdocs/webERP5/includes/ConnectDB_mariadb.inc:58
	Stack trace:
	#0 /home/.../webdocs/webERP5/includes/ConnectDB_mariadb.inc(58): mysqli_query()
	#1 /home/.../webdocs/webERP5/install/pages/page_6.php(203): DB_query()
	#2 /home/.../webdocs/webERP5/install/index.php(77): include('...')
	#3 {main}\n  thrown in /home/.../webdocs/webERP5/includes/ConnectDB_mariadb.inc on line 58, referer: https://erp.healthplus.local/install/index.php?Page=5
`
![Screenshot from 2024-12-28 12-26-30](https://github.com/user-attachments/assets/92f3f7ac-4c18-4e71-be27-a0d6b7a75f76)
The screenshot is where the install step stopped.  
Currently running under: Debian 12, PHP8.4, Apache2, MariaDB (fresh install).
Same error message for options mariadb, mysqli, with demo data, without demo data.

**Reply by pakricard** [reply]
Hi @bcinteractive :
I messed it up 2 days ago fixing #312, but ended up breaking this one. ;-)
I just uploaded the correct version, now the INSERT INTO tags goes on Page_6.php with the other ones.
I can't test it, please do and confirm us.

**Reply by bcinteractive** [reply]
Hi @pakricard :

- Updated my local copy of webERP master.
- The install procedure seems to be in order.
- After I hit the button for completing the install wizard, I'm redirected to the login page.
- After logging in with default account (admin/weberp) - met with a blank page and the error log message, see below

```
Got error 'PHP message: PHP Fatal error:  Uncaught mysqli_sql_exception: Table 'weberp_hp.login_data' doesn't exist in /home/.../webdocs/webERP5/includes/ConnectDB_mariadb.inc:58\n
	Stack trace:
	#0 /home/.../webdocs/webERP5/includes/ConnectDB_mariadb.inc(58): mysqli_query()
	#1 /home/.../webdocs/webERP5/includes/session.php(187): DB_query()
	#2 /home/.../webdocs/webERP5/index.php(4): include('...')
	#3 {main}
              thrown in /home/.../webdocs/webERP5/includes/ConnectDB_mariadb.inc on line 58', referer: https://erp.sample.local/index.php
```

**Comment by pakricard** [comment, marked as answer]
I think I located the root issue.

In the latest code there is a CREATE TABLE login_data in sql/updates/11.php but there is NO equivalent for a fresh install (no file in install/pages or install/tables takes action to create this login_data table.

That's why it does work for people upgrading their installation but can't work for fresh installations.

I created an install/tables/login_data.sql that should fix it.

**Comment by andrewcouling** [comment]
Does this code look correct for login_data.sql ?

```
CREATE TABLE `login_data` (
  `sessionid` char(32) DEFAULT NULL,
  `userid` varchar(20) DEFAULT NULL,
  `login` TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP,
  `script` varchar(100) NOT NULL DEFAULT '',
  KEY `sessionid` (`sessionid`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb3
```

**Comment by pakricard** [comment]
Hi @andrewcouling:

I committed install/tables/login_data.sql table that was a copy of the same SQL used by sql/updates/11.php, so a fresh start installation or an old one being updated should result with the same table:
 
CREATE TABLE `login_data` (
  `sessionid` char(26),
  `userid` varchar(20),
  `login` TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP,
  `script` varchar(100) NOT NULL DEFAULT "",
  PRIMARY KEY (`sessionid`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb3

In your version maybe the "PRIMARY" should be added to KEY, 
Thanks for spotting the difference in session field size ;-)
R

**Comment by andrewcouling** [comment]
Thanks @pakricard . 
I can now login after a fresh install so this is solved.
